### PR TITLE
Add a protocol for websocket

### PR DESCRIPTION
**Pull request by ostcar**
{'type': STRING, 'content': ANY}

### CODE REVIEW

File: openslides/utils/consumers.py [hunk 1]
```diff
@@ -112,7 +121,7 @@ class SiteConsumer(AsyncJsonWebsocketConsumer):
                 collection_string=collection_string,
                 id=id,
                 action='deleted'))
-        await self.send_json(output)
+        await self.send_json({'type': 'autoupdate', 'message': output})
```
Comment (FinnStutzenstein) [inline openslides/utils/consumers.py]: `content`

File: openslides/utils/consumers.py [hunk 2]
```diff
@@ -48,27 +48,36 @@ class SiteConsumer(AsyncJsonWebsocketConsumer):
...
+                    "site",
+                    {
+                        "type": "send_notify",
+                        "incomming": content,
+                        "senderReplyChannelName": self.channel_name,
+                        "senderUserId": self.scope['user'].id or 0,
+                    },
+                )
+            else:
+                await self.send_json({'error': 'invalid notify message'})
```
Comment (FinnStutzenstein) [inline openslides/utils/consumers.py]: Use the 'Protocol' for errors, too.

File: openslides/utils/consumers.py [hunk 3]
```diff
@@ -48,27 +48,36 @@ class SiteConsumer(AsyncJsonWebsocketConsumer):
...
         else:
-            await self.send_json({'error': 'invalid message'})
+            await self.send_json({'error': 'Unknown type {}'.format(content_type)})
```
Comment (FinnStutzenstein) [inline openslides/utils/consumers.py]: Same here. Maybe a `send_error(str)` is good.

File: openslides/utils/consumers.py [hunk 4]
```diff
@@ -48,27 +120,30 @@ class SiteConsumer(AsyncJsonWebsocketConsumer):
...
+                        "incomming": content,
+                        "senderReplyChannelName": self.channel_name,
+                        "senderUserId": self.scope['user'].id or 0,
+                    },
+                )
+            else:
+                await self.send_json(type='error', content='Invalid notify message', in_response=id)
         else:
-            await self.send_json({'error': 'invalid message'})
+            await self.send_json(type='error', content='Unknown type {}'.format(type), in_response=id)
```
[the remark this answers is not in the document]
Comment (ostcar) [reply openslides/utils/consumers.py]: done

File: openslides/utils/consumers.py [hunk 5]
```diff
@@ -148,25 +223,28 @@ class ProjectorConsumer(AsyncJsonWebsocketConsumer):
...
+                "site",
+                {
+                    "type": "send_notify",
+                    "incomming": content,
+                    "senderReplyChannelName": self.channel_name,
+                    "senderProjectorId": projector_id,
+                },
+            )
+        else:
+            await self.send_json(type='error', content='Unknown type {}'.format(type), in_response=id)
```
[the remark this answers is not in the document]
Comment (ostcar) [reply openslides/utils/consumers.py]: done

**Pull request merged**